### PR TITLE
Add transformer example

### PR DESCRIPTION
**Pull request by yuzisun**
**What this PR does / why we need it**:
- Add an example for pytorch serving with transformer! 
- fixed a bug where predictor host should include namespace 

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

**Release note**:

```release-note

```

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/kfserving/354)

### CODE REVIEW

**Comment by yuzisun** [comment]
/assign @animeshsingh 
cc @ellis-bigelow @cliveseldon @rakelkar

File: pkg/controller/kfservice/resources/knative/service.go [hunk 1]
```diff
@@ -209,9 +209,10 @@ func (c *ServiceBuilder) CreateTransformerService(name string, metadata metav1.O
 		annotations[autoscaling.ClassAnnotationKey] = autoscaling.KPA
 	}
 
-	predictorHostName := constants.DefaultPredictorServiceName(metadata.Name)
+	predictorHostName := fmt.Sprintf("%s.%s", constants.DefaultPredictorServiceName(metadata.Name), metadata.Namespace)
+
 	if isCanary {
-		predictorHostName = constants.CanaryPredictorServiceName(metadata.Name)
+		predictorHostName = fmt.Sprintf("%s.%s", constants.CanaryPredictorServiceName(metadata.Name), metadata.Namespace)
```
Comment (ellis-bigelow) [inline pkg/controller/kfservice/resources/knative/service.go]: Nice fix!

**Comment by yuzisun** [comment]
> Super cool! I'm having a hard time loading the ipynb in the github ui. Will try again last, but make sure it's uploaded correctly.

hmm I can not open it neither from GitHub but I can run the notebook fine locally, maybe I should remove the outputs on the notebook?

**Comment by yuzisun** [comment]
@ellis-bigelow  looks like it is working now on GitHub UI, can you try ? seems there is a batch job to render the ui which has delays :)

File: docs/samples/transformer/image_transformer/README.md [hunk 2]
```diff
@@ -0,0 +1,105 @@
+# Predict on a KFService using PyTorch Server and Transformer
+
+Most of model servers expect tensors as input data, so a pre-process step is needed before making the prediction call if user is sending in raw input format. Transformer is
+a service we orchestrated from KFService spec for user implemented pre/post process code. In the [pytorch](../../pytorch/README.md) example we send the prediction endpoint with
+tensor input in this example we add additional pre-process step to allow user sending in raw image data.
```
Comment (animeshsingh) [inline docs/samples/transformer/image_transformer/README.md]:
```suggestion
Most of the model servers expect tensors as input data, so a pre-processing step is needed before making the prediction call if the user is sending in raw input format. Transformer is a service we orchestrated from KFService spec for user implemented pre/post processing code. In the [pytorch](../../pytorch/README.md) example we call the prediction endpoint with tensor inputs, and in this example we add additional pre-processing step to allow the user send raw image data.
```

File: docs/samples/transformer/image_transformer/README.md [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+
+## Setup
+1. Your ~/.kube/config should point to a cluster with [KFServing installed](https://github.com/kubeflow/kfserving/blob/master/docs/DEVELOPER_GUIDE.md#deploy-kfserving).
+2. Your cluster's Istio Ingress gateway must be network accessible.
+3. Your cluster's Istio Egresss gateway must [allow Google Cloud Storage](https://knative.dev/docs/serving/outbound-network-access/)
+
+##  Build transformer image
```
Comment (animeshsingh) [inline docs/samples/transformer/image_transformer/README.md]:
```suggestion
##  Build Transformer image
```

File: docs/samples/transformer/image_transformer/README.md [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+
+### Extend Transformer and implement pre/postprocess functions
```
Comment (animeshsingh) [inline docs/samples/transformer/image_transformer/README.md]:
```suggestion
### Extend Transformer and implement pre/post processing functions
```

File: docs/samples/transformer/image_transformer/README.md [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+class ImageTransformer(Transformer):
+
+    def preprocess(self, inputs: Dict) -> Dict:
+        return {'instances': [image_transform(instance) for instance in inputs['instances']]}
+
+    def postprocess(self, inputs: List) -> List:
+        return inputs
+```
+
+### Build transformer docker image
```
Comment (animeshsingh) [inline docs/samples/transformer/image_transformer/README.md]:
```suggestion
### Build Transformer docker image
```

Comment (animeshsingh) [reply docs/samples/transformer/image_transformer/README.md]: We should also hyperlink to the Transformer file here in README, and to the Dockerfile as well

File: docs/samples/transformer/image_transformer/README.md [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+This step can be part of your CI/CD pipeline to continuously build the transformer image version. 
+```shell
+docker build -t yuzisun/image-transformer:latest -f transformer.Dockerfile .
+```
+
+## Create the KFService
+
```
Comment (animeshsingh) [inline docs/samples/transformer/image_transformer/README.md]:
```suggestion
Please use the [YAML file](docs/samples/transformer/image_transformer/image_transformer.yaml) to create the KFService, which includes a Transformer and a Predictor
```

File: docs/samples/transformer/image_transformer/image_transformer/__main__.py [hunk 7]
```diff
@@ -0,0 +1,32 @@
+# Copyright 2019 kubeflow.org.
```
Comment (animeshsingh) [inline docs/samples/transformer/image_transformer/image_transformer/__main__.py]: Not sure if we need two files for transformer code? May be main can be combined into the 2nd file for simplicity?

**Pull request merged**